### PR TITLE
Switch Docker images using musl based changelog-tool

### LINKED ISSUE #35: Update dockerfile to use musl
For the Dockerfile, we want a multistage build that uses the musl version of ponyc to build the changelog-tool and then we can copy over that build artifact into a new alpine based image.

It will end up being smaller than the current image.

### PR DESCRIPTION
**Pull request by SeanTAllen**
It will be more portable across Linux environments when we want to
grab from an image for use in other containers.

Closes #35

### CODE REVIEW

**Comment by SeanTAllen** [comment]
This shouldn't be merged until #36 is merged so that the changelog will be automatically updated.

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-  && make install \
-  && make clean
+ && make install
 
-WORKDIR /src/main
+FROM alpine:3.10
+
+COPY --from=build /usr/local/bin/changelog-tool /usr/local/bin/changelog-tool
 
 CMD changelog-tool
```
Comment (EpicEric) [inline Dockerfile:20]: Getting the following errors when running the image due to missing dependencies in the alpine image:

```
Error loading shared library libatomic.so.1: No such file or directory (needed by /usr/local/bin/changelog-tool)
Error loading shared library libgcc_s.so.1: No such file or directory (needed by /usr/local/bin/changelog-tool)
[...]
```

This is fixed by:

```diff
diff --git a/Dockerfile b/Dockerfile
index 7f8c695..15fc081 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -12,4 +12,6 @@ FROM alpine:3.10
 
 COPY --from=build /usr/local/bin/changelog-tool /usr/local/bin/changelog-tool
 
+RUN apk add --update libatomic gcc
+
 CMD changelog-tool
```

**Review by EpicEric — CHANGES_REQUESTED** [review]

**Comment by SeanTAllen** [comment]
Ah, there's a better fix. I need to address in the make command.

**Comment by EpicEric** [comment]
If you moved the Pony files to `changelog-tool/`, it's being ignored in `.gitignore`.

**Comment by SeanTAllen** [comment]
lol thanks @EpicEric, that would have taken me forever to find

**Comment by SeanTAllen** [comment]
@EpicEric that's much better. 

can you give it another review?

am i doing the docker parts correctly?

File: Dockerfile (hunk 1 above)
Comment (SeanTAllen) [reply Dockerfile:20]: addressed through a lot more work.

Comment (SeanTAllen) [reply Dockerfile:20]: This should be fixed via a lot more work.

**Review by EpicEric — APPROVED** [review]

**Pull request merged**